### PR TITLE
Use signed URLs instead of DcaFilterInterface to forward DCA configuration

### PR DESCRIPTION
**Pull request by aschempp**
As discussed

- removed use of DcaFilterInterface
- the Interface file still exists for BC reasons
- config for pageTree and fileTree widgets are supported, but for no other

I still want to discuss the `blnHideFiles` etc. props in DC_Folder. I think they're incorrectly implemented and the code was hard to adjust.

### CODE REVIEW

**Comment by aschempp** [comment]
Unfortunately, this breaks as soon as someone does anything with the URL, like toggle the tree 😢

File: src/DataContainer/DcaFilterInterface.php [hunk 1]
```diff
@@ -14,6 +14,8 @@
  * DCA filter interface.
  *
  * @author Leo Feyer <https://github.com/leofeyer>
+ *
+ * @deprecated Deprecated since Contao 4.4.1, to be removed in Contao 5.
```
Comment (leofeyer) [inline src/DataContainer/DcaFilterInterface.php]: Since the interface has only been added in version 4.4.0 and the functionality is broken, we can remove it completely.

File: src/Menu/FilePickerProvider.php [hunk 2]
```diff
@@ -103,6 +105,10 @@ public function processSelection($value)
      */
     public function canHandle(Request $request)
     {
+        if ($request->query->get('context') === 'file') {
```
Comment (leofeyer) [inline src/Menu/FilePickerProvider.php]: Why did you add this?

**Comment by leofeyer** [comment]
> Unfortunately, this breaks as soon as someone does anything with the URL, like toggle the tree 😢 

Does this mean that we cannot use this approach at all?

**Comment by aschempp** [comment]
> Does this mean that we cannot use this approach at all?

Yeah, I guess so :(

**Comment by Toflar** [comment]
We have a few ideas here, working on a solution :)
